File: schemaorggo/borrow_action.go
package schemaorggo

import "encoding/json"

// BorrowAction see : https://schema.org/BorrowAction
type BorrowAction struct {
	TransferAction

	typeContext

	// Lender see : https://schema.org/lender
	// A sub property of participant. The person that lends the object being borrowed.
	// types : Organization Person
	Lender []interface{} `json:"lender,omitempty"`
}

func (v BorrowAction) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.TransferAction.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.Lender
		if len(v.Lender) == 1 {
			value = v.Lender[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["lender"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v BorrowAction) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "BorrowAction"

	return data, nil
}

func (v BorrowAction) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/borrow_action.php
<?php

class BorrowAction extends TransferAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'BorrowAction';
	
	/**
	 * A sub property of participant. The person that lends the object being borrowed.
	 * see : https://schema.org/lender
	 * @var \Organization|\Organization[]|\Person|\Person[]
	 */
	public var $lender;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'BorrowAction'
		);
		
		$serialized = so_json_serialize( $this->lender );
		if ( ! empty( $serialized ) ) {
			$out['lender'] = $serialized;
		}
		
		return $out;
	}
}
